x.consolidation/consolidation.md

```diff
@@ -0,0 +1,517 @@
+That's because saving a few dozen bytes in block chain space using
+predictive/organic transactions doesn't match being able to save a large
+percentage of UTXO spending costs by spending UTXOs at a lower feerate.
+This was actually quantified in a earlier illustration: 
+
+![Savings by consolidation feerate](img/estimating-savings-by-feerate.png)
+
+For those that spend an average of two UTXOs or more per transaction
+and who can consolidate at 50% or less of their normal feerate (and who
+use the P2WPKH transactions analyzed here), consolidation at least
+breaks even on costs and can easily save more money than sending a
+normal transaction.  The feerate data included in the appendix to this
+chapter shows that saving at least 50% on feerates for a nominal 2-block
+confirmation target was possible 99.85% of the time during our analysis
+period of November and December 2019.
+
+Again, the predictive/organic method does have the advantage that it may
+be significantly easier to implement than consolidation for existing
+services, and combined with changeless transactions it may achieve a
+modest additional savings plus boost privacy.  Still, spenders focused
+on maximizing fee savings will probably want to focus on UTXO consolidation.
+
+## Fee bumping
+
+The maximize savings, consolidation transactions should be sent at the
+lowest feerate that you believe will allow the transactions to confirm
+before they are needed.  For example, if you can wait a week for a
+transaction to confirm, you should use your fee estimator's target
+feerate for confirmation within 1,008 blocks.
+
+However, sometimes you may need to fee bump a previously-sent
+consolidation transaction in order to make it confirm faster or because
+feerates have risen unexpectedly.  In that case, you have several
+options:
+
+- For consolidations from a hot wallet to a hot wallet (or from a hot
+  wallet to a cold wallet), you should signal [BIP125][] so that you can
+  use Replace-by-Fee (RBF) fee bumping, which is the most efficient type of
+  fee bumping.
+
+- For consolidations sent from a cold wallet to a hot wallet, you can
+  easily use Child-Pays-For-Parent (CPFP) fee bumping.
+
+- For consolidations sent from a cold wallet to a cold wallet, you may
+  want to design your transactions so that they always include a small
+  output paid to your hot wallet.  This will allow you to use CPFP fee
+  bumping via the hot wallet without needing to re-open the cold wallet.
+  However, this extra output may slightly reduce the efficiency of your
+  consolidation unless you were planning to transfer funds to your hot
+  wallet anyway.
+
+## Conclusion 
+
+For organizations that both frequently receive and frequently send
+payments, consolidation is one of the most efficient ways to save money.
+It's relatively easy to implement and, if done well, the people
+receiving payments from you will never notice that you're doing it.
+
```

Comment:
What about consolidating while spending (adding 'unnecessary' inputs to the current transaction which are consolidated into a larger change output). And even moving from hot to cold at the same time.

Reply:
Consolidating while spending is the Predictive/Organic UTXO Management technique.  I'll see if I can make that clearer in that section.

I think it should be pretty rare that a high-frequency spender moves funds from hot to cold---for security, they should be receiving new deposits/payments to their cold wallet and then only moving enough money to the hot wallet to cover a limited number of outgoing payments/withdrawals (e.g. a week's worth).  I think you'd only want to move from hot to cold if there was emergency or maybe a huge price increase.

Reply:
Perhaps this miss was just my lack of understanding of the predictive/organic utxo mgmt techniques.

Reply:
I did add a sentence to the first paragraph of the predictive/organic section to better describe how it works so hopefully other people won't encounter the same confusion.

Reply:
Resolved